### DISCUSSION TITLE
Cisco IOS as supported platform

### DISCUSSION THREAD

**Discussion by Kani999** [carlmontanari/scrapli#158] [General]
Hello, 

Are you considering adding the plain CISCO IOS platform as part of the supported platforms? 

https://scrapli.github.io/scrapli_cfg/user_guide/project_details/#supported-platforms

Best regards,
Jan Krupa

**Comment by carlmontanari** [comment]
Hey @Kani999 -- yes and no! 😁 

"Yes" because AFAIK the iosxe platform should work for any ios device. "No" because there is no "dedicated" platform for standard ios since the iosxe platform should work! If ya run into something that *doesnt* work -- please do raise an issue though so we can take a peak!

Carl

**Comment by salkindnul** [comment]
How about Cisco ASA Firewalls?

Niklas

**Reply by carlmontanari** [reply]
> How about Cisco ASA Firewalls?

The IOSXE platform probably works for at least some ASA things. Generic Driver will absolutely work but you are then on your own for privilege handling/prompt patterns/etc.. 

That said, it is fairly trivial to add device types in [scrapli community](https://github.com/scrapli/scrapli_community) though, so if there is any platform not present it can be added! In general adding a platform means understanding the "privilege levels" of that device (i.e. exec/privilege exec/config/etc.) and working through some regex bits to ensure that scrapli knows how to decide what privilege level its at. There are some docs/examples in the community repo on how to do this, and I can help with the PR if you decide to go that route.

Carl
